db/repos/blocks.js

```diff
@@ -23,110 +25,238 @@ var columnSet;
+	 * @param {string} id
+	 * @return {Promise}
+	 */
+	getGenesisBlockId (id) {
+		// TODO: Must use a result-specific method, not .query
+		return this.db.query(sql.getGenesisBlockId, [id]);
+	}
+
+	/**
+	 * Delete a block from database
+	 * @param {string} id
+	 * @return {Promise}
+	 */
+	deleteBlock (id) {
+		return this.db.none(sql.deleteBlock, [id]);
+	}
+
+	/**
+	 * Aggregate rewards for a block
+	 * @param {Object} params
+	 * @param {string} params.generatorPublicKey
+	 * @param {int} params.start - Start time of aggregation period
+	 * @param {int} params.end - End time for aggregation period
+	 * @return {Promise}
+	 */
+	aggregateBlocksReward (params) {
+		// TODO: Must use a result-specific method, not .query
+		return this.db.query(sql.aggregateBlocksReward, [params.generatorPublicKey, params.start, params.end]);
+	}
+
+	/**
+	 * Count blocks
+	 * @return {Promise}
+	 */
+	count () {
+		// TODO: Must use inline result conversion
+		return this.db.one(sql.count);
+	}
+
+	/**
+	 * Search blocks in database
+	 * @param {Object} params
+	 * @param {array} params.where
+	 * @param {string} params.sortField
+	 * @param {string} params.sortMethod
+	 * @param {int} params.limit
+	 * @param {int} params.offset
+	 * @return {Promise}
+	 */
+	list (params) {
+		// TODO: Must use a result-specific method, not .query
+		return this.db.query(Queries.list(params), params);
+	}
+
+	/**
+	 * Get sequence of blocks ids for delegates
+	 * @param {Object} params
+	 * @param {int} params.delegates - Number of delegates
+	 * @param {int} params.height
+	 * @param {int} params.limit
+	 * @return {Promise}
+	 */
+	getIdSequence (params) {
+		// TODO: Must use a result-specific method, not .query
+		return this.db.query(sql.getIdSequence, params);
+	}
+
+	/**
+	 * Get common block among peers
+	 * @param {Object} params
+	 * @param {string} params.id
+	 * @param {string} params.previousBlock
+	 * @param {int} params.height
+	 * @return {Promise}
+	 */
+	getCommonBlock (params) {
+		// TODO: Must use a result-specific method, not .query
+		params.comparePreviousBlock = params.previousBlock ? this.pgp.as.format('AND "previousBlock" = ${previousBlock}', params): '';
+		return this.db.query(sql.getCommonBlock, params);
+	}
+
+	// TODO: Merge BlocksRepository#getHeightByLastId with BlocksRepository#list
+	/**
+	 * Get height of the block with id
+	 * @param {string} lastId - Id of the block to search
+	 * @return {Promise}
+	 */
+	getHeightByLastId (lastId) {
+		// TODO: Must use a result-specific method, not .query
+		return this.db.query(sql.getHeightByLastId, [lastId]);
+	}
+
+	/**
+	 * Load block including transactions
+	 * @param {Object} params
+	 * @param {string} params.id
+	 * @param {string} params.lastId
+	 * @param {int} params.height
+	 * @param {int} params.limit
+	 * @return {Promise}
+	 */
+	loadBlocksData (params) {
+		// TODO: Must use a result-specific method, not .query
+		return this.db.query(Queries.loadBlocksData(params), params);
```

Comment:
Such dynamically generated queries are a challenge to handle in SQL but we have to came up with some solution.

Reply:
Same as per my previous comment, this does need to be done, but on individual basis. Possibly, joining the effort to throw away both those two cases that are left.